### PR TITLE
[WIP] on endpoints

### PR DESCRIPTION
**Pull request by psav**
Requires an update to the yaml to include something like the following

``` yaml
        endpoints:
            default:
                hostname: 10.0.0.0
                credentials: these_creds
```

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 1fb1501ff892fe0d155ded55187629ddebb857c1:

`cfme/infrastructure/provider/virtualcenter.py`:
- :red_circle: Line 4:1: **N814** *camelcase imported as constant*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 277986259c44ae91b24f56259925b62adf0448d4:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit dd3108d4325ab7a9c075c5e0910692053e16e856:

Everything seems all right :smile: .
*CFME QE Bot*

File: cfme/infrastructure/provider/virtualcenter.py [hunk 1]
```diff
@@ -1,5 +1,16 @@
 from mgmtsystem.virtualcenter import VMWareSystem
 from . import Provider
+from cfme.web_ui import Form, Input
+from cfme.common.provider_endpoint import ProviderEndpoint
+
+
+default_endpoint_form = Form(
```
Comment (jkrocil) [inline cfme/infrastructure/provider/virtualcenter.py]: I wonder if this should be defined inside a `ProviderEndpointDefault` class seeing as its closely tied to it. Same for other endpoint types. So `ProviderEndpointBase` would define the generic fill and classes like `ProviderEndpointCandU` and `ProviderEndpointDefault` would contain the fields definition and would potentially also know how to parse those fields from provider config.
```
endpoints = [
    ProviderEndpointDefault.from_config(prov_config),
    ProviderEndpointCandU.from_config(prov_config)]
```
Id say thats easy to understand, no magic happening there and its easy to maintain.
Thoughts?

Comment (jkrocil) [inline cfme/infrastructure/provider/virtualcenter.py]: That said, one should still be able to initialize an endpoint using k:v pairs and should be able to query the class for supported values easily.

Comment (psav) [inline cfme/infrastructure/provider/virtualcenter.py]: Yeh, I like that idea, close to what I was doing originally actually

**Comment by dajoRH** [comment]
Lint report for commit 134271dbb7f579c8dd3e468cf3b1a008e953ec6f:

Everything seems all right :smile: .
*CFME QE Bot*

File: cfme/common/provider.py [hunk 2]
```diff
@@ -196,8 +196,13 @@ def create(self, cancel=False, validate_credentials=True):
         """
         sel.force_navigate('{}_provider_new'.format(self.page_name))
         fill(self.properties_form, self._form_mapping(True, **self.__dict__))
-        for cred in self.credentials:
-            fill(self.credentials[cred].form, self.credentials[cred], validate=validate_credentials)
+        if hasattr(self, 'endpoints'):
```
Comment (psav) [inline cfme/common/provider.py]: This is just to maintain backwards compat for now whilst testing

**Comment by dajoRH** [comment]
Lint report for commit f4e3710aa84729d09a75e0a4403f490d27dcc094:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 10f5e6edb2d6640be2de02241bae05423242c4c0:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit e0f178e6971eb5b71be82a42ef54206528719f9c:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 8accf7ea293c0f3ef2f1e91c87a1fef9785d217a:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 00e63900480cc96b22bcd3b03802f5215ff71cec:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by izapolsk** [comment]
@psav, I'm currently fighting with endpoints. I'm really interested in having this merged. Can I help you somehow with it ?

**Pull request closed without merging**